src/openforms/formio/dynamic_config/referentielijsten.py

```diff
@@ -68,4 +68,14 @@ def fetch_options_from_referentielijsten(
                 component,
                 _("No results found from Referentielijsten API."),
             )
-        return [[item["code"], item["naam"]] for item in result]
+            return
+
+        valid_items = []
+        for item in result:
+            if (
+                einddatum := item.get("einddatumGeldigheid")
+            ) and check_expired_or_near_expiry(einddatum):
+                continue
+
+            valid_items.append(item)
+        return [(item["code"], item["naam"]) for item in valid_items]
```

Comment:
```suggestion
        return [
            (item["code"], item["naam"]) for item in valid_items
            if not (einddatum := item.get("einddatumGeldigheid")) or not check_expired_or_near_expiry(einddatum)
        ]
```
either this or just push the tuples directly onto `valid_items`, now you're looping twice over a list of items which could be done in a single pass